jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -96,12 +101,25 @@ protected Selector newSelector() throws IOException
         return Selector.open();
     }
 
+    public Selector getSelector()
+    {
+        return _selector;
+    }
+
+    protected void onSelectFailed(Throwable cause)
+    {
+        LOG.warn(toString(), cause);
```

Comment:
I think we should restart the selector by default here.
If the selector has failed, then we have a dead server - unless somehow a selector restart happens. 
By restarting here we risk only that we might get into a restart loop - which again gives us a dead server, but one eating 100% CPU so it will be more obvious than one that dropped a selector.

For good servers running on proper OSs, the selector never fails in this way, so this is never a problem.  So I see little risk in restarting here.... if we ever do get a problem with restart loops, we could consider a restart quota.